ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -220,48 +211,41 @@ View getBannerView() {
     return adLayout;
   }
 
-  public void onDestroy() {
+  void onDestroy() {
     Log.d(TAG, "onDestroy: " + hashCode());
-    destroy(adLayout);
-    adLayout = null;
+    VungleManager.getInstance().destroyBannerAd(this);
   }
 
-  public void onPause() {
+  void onPause() {
     Log.d(TAG, "onPause");
     updateVisibility(false);
   }
 
-  public void onResume() {
+  void onResume() {
     Log.d(TAG, "onResume");
     updateVisibility(true);
   }
 
+  @NonNull
+  String getPlacementId() {
+    return mPlacementId;
+  }
+
   @Nullable
   String getUniquePubRequestId() {
     return mUniquePubRequestId;
   }
 
   boolean isActive() {
-    return adLayout != null;
-  }
-
-  void destroy(@Nullable View adView) {
-    Log.d(TAG, "Vungle banner adapter try to destroy:" + this);
-    if (adView == adLayout) {
-      Log.d(TAG, "Vungle banner adapter destroy:" + this);
-      mVisibility = false;
-      VungleManager.getInstance().removeActiveBannerAd(mPlacementId);
-      cleanUp();
-      mPendingRequestBanner = false;
-    }
+    return adapter.get() != null;
```

Comment:
"adapter" is leaking

Reply:
yes, after fixed bannerListener adapter no-leak on my side.